tests/shared/importers/test_importer.py

```diff
@@ -360,5 +360,5 @@ async def test_nlu_data_domain_sync_with_retrieval_intents(project: Text):
 
     assert domain.retrieval_intents == ["chitchat"]
     assert domain.intent_properties["chitchat"].get("is_retrieval_intent")
-    assert domain.templates == nlu_data.responses
+    assert domain.retrieval_intent_templates == nlu_data.responses
```

Comment:
This test is using `data/test_domains/default_retrieval_intents.yml` that doesn't have any responses for the retrieval intents 😬 So the test is comparing empty lists. Can we probably change it?

I'd also keep one more check:
```python
assert domain.templates != nlu_data.responses
```

Reply:
It's actually not empty lists because, the NLU data file used(`data/test_responses/default.md`) has responses for retrieval intents defined. So, when a 2-way sync happens inside `RetrievalModelsDataImporter`, both `domain.retrieval_intent_templates` and `nlu_data.responses` should have the same response templates.
I agree `assert domain.templates != nlu_data.responses` is a good check to have there.

Reply:
Although, `assert domain.templates != nlu_data.responses` is not valid if your assistant has only retrieval intent templates and no other response templates. So, the real check is that `domain.retrieval_intent_templates == nlu_data.responses` and `domain.retrieval_intent_templates` is a proper subset of `domain.templates`. What do you think?

Reply:
> NLU data file used(data/test_responses/default.md)

Right, I think I missed the point that it's a both way sync happening 😄 
My idea is - since we're modifying the domain file's logic, to test the domain, not the syncing.
So I'd introduce a domain file with both non-retrival-intents responses AND retrival-intents responses, and then check that `domain.templates != domain.retrieval_intent_templates` and both are non-empty.
Does that make sense?

Reply:
⏫ And that should ideally be a pure domain test, without any sync with NLU data.

Reply:
That makes sense. Added an additional test for this.